Sources/HTTPKit/Utilities/HTTPHeaderValue.swift

```diff
@@ -197,3 +197,26 @@ public struct HTTPHeaderValue {
         return HTTPHeaderValue(.init(value), parameters: parameters)
     }
 }
+
+extension HTTPHeaderValue: Codable {
+    /// Initialize a HTTPHeaderValue from a Decoder
+    ///
+    /// This will decode a String from the decoder and parse it
+    public init(from decoder: Decoder) throws {
+        let container = try decoder.singleValueContainer()
+        let string = try container.decode(String.self)
+        guard let tempValue = HTTPHeaderValue.parse(string) else {
+            throw DecodingError.dataCorruptedError(in: container, debugDescription: "Invalid header value string")
+        }
+        self.parameters = tempValue.parameters
+        self.value = tempValue.value
+    }
+    
+    /// Encode a HTTPHeaderValue into an Encoder
```

Comment:
type names in doc blocks should be back-ticked to make them clickable in generated docs

Reply:
Will fix!